### PR TITLE
Ensure that iterating over the library map doesn't trigger pointer underflow

### PR DESCRIPTION
**Pull request by daniel-grumberg**
This patch introduces a safe version of the `hlist_entry` macro like this [one](https://elixir.bootlin.com/linux/latest/source/include/linux/list.h#L856)

This also includes some reformatting and typo fixes.

### CODE REVIEW

File: cmake/c_flags_override.cmake [hunk 1]
```diff
@@ -3,3 +3,4 @@ set(CMAKE_C_FLAGS_DEBUG_INIT "-O0 -ggdb3")
 set(CMAKE_C_FLAGS_MINSIZEREL_INIT "-Os")
 set(CMAKE_C_FLAGS_RELEASE_INIT "-O3")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO_INIT "-O2 -ggdb3")
+set(CMAKE_C_STANDARD 99)
```
Comment (andronat) [inline cmake/c_flags_override.cmake]: I think Varan requires ‘C11’. Do you think this might create issues?

**Comment by ccadar** [comment]
Nice find, @daniel-grumberg !  
But yes, why C99?

**Comment by daniel-grumberg** [comment]
@parras will do

**Comment by daniel-grumberg** [comment]
@ccadar @andronat I saw we were using bool quite a lot which was only introduced in C99. Varan being C11 shouldn’t be a problem, as the ABI is the same, but I might as well make it C11 for everyone.

**Pull request merged**